### PR TITLE
JUnit5 migration

### PR DESCRIPTION
**Pull request by 0x006EA1E5**
Addresses #907

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -80,75 +80,87 @@
...
+                <scope>import</scope>
+            </dependency>
+        </dependencies>
+    </dependencyManagement>
+
     <build>
         <pluginManagement>
             <plugins>
                 <!-- Add plugin versions for reproducible builds. Versions for default bindings are taken from -->
-                <!-- https://github.com/apache/maven/tree/maven-3.8.1/maven-core/src/main/resources/META-INF/plexus -->
```
Comment (0x006EA1E5) [inline pom.xml:87]: Is this change okay in this PR?  I really only needed to update surefire/failsafe...

File: pom.xml [hunk 2]
```diff
@@ -213,6 +225,7 @@
                     <show>public</show>
                     <skippedModules>benchmarks,examples,integration-tests,integration_tests,,simpleclient,simpleclient_bom,simpleclient_caffeine,simpleclient_common,simpleclient_dropwizard,simpleclient_graphite_bridge,simpleclient_guava,simpleclient_hibernate,simpleclient_hotspot,simpleclient_httpserver,simpleclient_jetty,simpleclient_jetty_jdk8,simpleclient_log4j,simpleclient_log4j2,simpleclient_logback,simpleclient_pushgateway,simpleclient_servlet,simpleclient_servlet_common,simpleclient_servlet_jakarta,simpleclient_spring_boot,simpleclient_spring_web,simpleclient_tracer,simpleclient_vertx,simpleclient_vertx4</skippedModules>
                     <excludePackageNames>io.prometheus.metrics.expositionformats.generated.*</excludePackageNames>
+                    <sourcepath>src/main/java;src/main/generated</sourcepath>
```
Comment (fstab) [inline pom.xml:228]: This line has been added here. Is that on purpose? Seems unrelated to JUnit.

**Review by fstab — CHANGES_REQUESTED** [review]
Hi Greg, thanks a lot for updating JUnit, and sorry for the delay. Looks good, I just have commented a small question on one line :)

File: pom.xml (hunk 2 above)
Comment (0x006EA1E5) [reply pom.xml:228]: This was needed to get `./mvnw javadoc:jar` to work.  Probably because I updated the version of `maven-javadoc-plugin`.

I have tried to compare the generated javadocs and they look substantially the same, with only minor differences in how the navigation works.  I see no differnces in the content.

To check, I cloned the repo to `client_java_copy` and did `./mvnw javadoc:jar` in both dirs to generate the javadocs.
Then I did `for file in **/target/apidocs; do diff "$file" "../client_java_copy/${file}" >> diff.txt; done` to capture the diffs to 
[diff.txt](https://github.com/prometheus/client_java/files/13754454/diff.txt)

Comment (0x006EA1E5) [reply pom.xml:228]: @fstab Is there anything more you you need for this?

Comment (fstab) [reply pom.xml:228]: Sorry for dropping the ball here, I put it back on my TODO list and will get back to you soon :crossed_fingers:.

Comment (0x006EA1E5) [reply pom.xml:228]: @fstab any progress?

File: pom.xml [hunk 3]
```diff
@@ -80,75 +80,87 @@
         </repository>
     </distributionManagement>
 
+    <dependencyManagement>
+        <dependencies>
+            <dependency>
+                <groupId>org.junit</groupId>
+                <artifactId>junit-bom</artifactId>
+                <version>5.10.1</version>
```
Comment (mimaison) [inline pom.xml:88]: 5.11.1 is out now

File: integration-tests/it-common/src/test/java/io/prometheus/client/it/common/Volume.java [hunk 4]
```diff
@@ -24,7 +24,7 @@ private Volume(Path tmpDir) {
 
     public static Volume create(String prefix) throws IOException, URISyntaxException {
         Path targetDir = Paths.get(Volume.class.getResource("/").toURI()).getParent();
-        Assert.assertEquals("failed to locate target/ directory", "target", targetDir.getFileName().toString());
+        Assertions.assertEquals("target", targetDir.getFileName().toString(), "failed to locate target/ directory");
```
[the remark this answers is not in the document]
Comment (zeitlinger) [reply integration-tests/it-common/src/test/java/io/prometheus/client/it/common/Volume.java:27]: yes, and assertj would also be great

File: mvnw [hunk 5]
```diff
@@ -19,15 +19,14 @@
 # ----------------------------------------------------------------------------
 
 # ----------------------------------------------------------------------------
-# Maven Start Up Batch script
+# Apache Maven Wrapper startup batch script, version 3.2.0
```
[the remark this answers is not in the document]
Comment (zeitlinger) [reply mvnw:22]: I don't mind

[the remark this answers is not in the document]
Comment (0x006EA1E5) [reply mvnw:22]: Hey guys, I thought this PR had been forgotten about.

Do you need me to look at these issues?

File: integration-tests/it-common/src/test/java/io/prometheus/client/it/common/Volume.java (hunk 4 above)
[the remark this answers is not in the document]
Comment (zeitlinger) [reply integration-tests/it-common/src/test/java/io/prometheus/client/it/common/Volume.java:27]: fixed in https://github.com/prometheus/client_java/pull/1092

**Comment by zeitlinger** [comment]
@0x006EA1E5 junit 5 migration is done - please add anything missing in a new pr
